### ISSUE TITLE
Why is there no dropout in the code?

### ISSUE THREAD

**Issue by wizardk** [janvainer/speedyspeech#35]
There is no dropout in the code just because the model is too small to over-fitting?

**Comment by janvainer** [comment]
Hi, good question. We did not put this in the paper, but we dedicated a paragraph to dropout and batch norm in my [thesis](https://is.cuni.cz/webapps/zzp/download/120361940/?lang=cs).
The relevan paragraph is below. In short, dropout destabilized inference and worsened the results. But feel free to play with it more, maybe there is some setting that will improve generalization.

> batch normalization. The batch normalization decreased the necessary training time.We also tried applying channel normalization without considering the batch dimension, but this type of normalization provided slightly worse results and did not provide significant stability gains during inference.We experimented with dropout applied after batch normalization and ReLU in all layers to encourage generalization.  However, this approach reduced the model capacity which was already fully used (more on that later), and also made the network very unstable during inference – the duration predictor would predict shorter durations for most of the phonemes which in turn negatively influenced the synthesized spectrograms. We suspect that dropout causes inconsistency in the variance of the intermediate network states.  However, more experiments would be needed to confirm this hypothesis

**Comment by wizardk** [comment]
Thanks for your illustration. I think data augmentation is as important as avoiding over-fitting. I'll try more experiments on your great work and I'll let you know if I got any improvements.
